### PR TITLE
Thermostat Capability - closes #13

### LINKED ISSUE #13: Thermostat Capability Schema
See also mozilla-iot/wot#61

### PR DESCRIPTION
**Pull request by benfrancis**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -847,6 +901,34 @@ <h2>AlarmProperty</h2>
...
+          </tr>
+          <tr>
+            <th>enum</th>
+            <td>["off", "heating", "cooling"]</td>
+          </tr>
+        </table>
+      </section>
+      <section id="ThermostatModeProperty">
+        <h2>ThermostatModeProperty</h2>
+        <p>Defines the current setting of a thermostat type device which can switch on and off a heating device and/or cooling device based on a current and desired temperature. Must support an "off" state and must support one or more of a "heat" state, "cool" state and "auto" state, where "auto" means both "heat" and "cool".</p>
```
Comment (mrstegeman) [inline index.html]: Instead of `both "heat" and "cool"`, maybe something like `either "heat" or "cool", depending on target and current temperatures`.

**Comment by benfrancis** [comment]
@dhylands wrote:
> So this seems to have some notion of heating/cooling, but only seems to have a single target temperature, rather than a separate heating target temperature and cooling target temperature.

Yeah I wasn't exactly sure how to deal with this. My thinking was that you could just have two `TargetTemperatureProperty` properties in one device.

```
{
  "properties": {
    "targetCoolingTemperature: {
      "@type": "TargetTemperatureProperty",
      "type": "number",
      "unit": "degree celsius",
      "title": "Target Cooling Temperature"
    },
    "targetHeatingTemperature: {
      "@type": "TargetTemperatureProperty",
      "type": "number",
      "unit": "degree celsius",
      "title": "Target Heating Temperature"
    }
  }
}
```

Do you think there should be a separate `HeatingTargetTemperature` and `CoolingTargetTemperature` property types?

> The auto ThermostatModeProperty makes no sense with only a single target temperature.

Are you sure it's definitely not possible to have a thermostat with only a single target temperature, but which does both heating and cooling? I'm sure I've seen this before, and assumed that it just maintains the target temperature within a certain margin.

If this is the case then that would neither be both a `HeatingTargetTemperature` `CoolingTargetTemperature`. Would it be less confusing to just have one or more `TargetTemperature` properties?

**Comment by mrstegeman** [comment]
I think it makes the most sense to just have `TargetTemperatureProperty`, but maybe make a note that a device may include multiple of them to account for different heating/cooling targets.

File: index.html [hunk 2]
```diff
@@ -917,7 +917,7 @@ <h2>HeatingCoolingProperty</h2>
       </section>
       <section id="ThermostatModeProperty">
         <h2>ThermostatModeProperty</h2>
-        <p>Defines the current setting of a thermostat type device which can switch on and off a heating device and/or cooling device based on a current and desired temperature. Must support an "off" state and must support one or more of a "heat" state, "cool" state and "auto" state, where "auto" means both "heat" and "cool".</p>
+        <p>Defines the current setting of a thermostat type device which can switch on and off a heating device and/or cooling device based on a current and desired temperature. Must support an "off" state and must support one or more of a "heat" state, "cool" state and "auto" state, where "auto" means either "heat" or "cool" depending on depending on target and current temperatures.</p>
```
Comment (mrstegeman) [inline index.html]: s/depending on depending on/depending on/

**Pull request merged**